src/qibolab/instruments/rfsoc.py

```diff
@@ -321,25 +317,29 @@ def play(
         return results
 
     @staticmethod
-    def validate_input_command(sequence: PulseSequence, execution_parameters: ExecutionParameters):
-        """Checks if sequence and execution_parameters are supported"""
+    def validate_input_command(sequence: PulseSequence, execution_parameters: ExecutionParameters, sweep: bool):
+        """Check if sequence and execution_parameters are supported."""
         if any(pulse.duration < 10 for pulse in sequence):
             raise ValueError("The minimum pulse length supported is 10 ns")
         if execution_parameters.acquisition_type is AcquisitionType.RAW:
-            raise NotImplementedError("Raw data acquisition is not supported")
+            if sweep:
+                raise NotImplementedError("Raw data acquisition is not compatible with sweepers")
+            if len(sequence.ro_pulses) != 1:
+                raise NotImplementedError("Raw data acquisition is compatible only with a single readout")
+            if execution_parameters.averaging_mode is not AveragingMode.CYCLIC:
+                raise NotImplementedError("Raw data acquisition can only be averaged")
```

Comment:
Have you tested if it only allows a single waveform to be acquired ? Maybe it is useful if 2 fit i memory for readout crosstalk analysis ?

Reply:
Thanks @Jacfomg, you are right and I'll try!
I could probably add something more sofisticated than just counting the readout pulses... In fact also single readout pulse, very long, could fill the memory...
In any case, I think all of these are just general improvements and not really required in the short term